pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -414,6 +459,39 @@ func (r *Reconciler) getCustomUserData() (string, error) {
 	return string(data), nil
 }
 
+// Return the value of the `sysprep-specialize-script-ps1` key from the provider spec metadata
+// or an empty string if not found.
+func (r *Reconciler) getWindowsScriptFromProviderSpec() string {
+	script := ""
+
+	for _, item := range r.providerSpec.Metadata {
+		if item.Key == windowsScriptMetadataKey {
+			script = windows.RemovePowershellTags(*item.Value)
+			break
+		}
+	}
+
+	return script
+}
+
+func (r *Reconciler) getWindowsScriptFromSecret() (string, error) {
+	var scriptSecret corev1.Secret
+	if err := r.coreClient.Get(context.Background(), client.ObjectKey{Namespace: windowsScriptSecretNamespace, Name: windowsScriptSecretName}, &scriptSecret); err != nil {
+		if apimachineryerrors.IsNotFound(err) {
+			return "", machinecontroller.InvalidMachineConfiguration("Windows script data secret %q in namespace %q not found: %v", windowsScriptSecretName, windowsScriptSecretNamespace, err)
+		}
+		return "", fmt.Errorf("error getting Windows script data secret %q in namespace %q: %v", windowsScriptSecretName, windowsScriptSecretNamespace, err)
+	}
+
+	data, exists := scriptSecret.Data[windowsScriptMetadataKey]
```

Comment:
The `sysprep-specialize-script-ps1` key may not be available if the script was loaded from a secret [created](https://github.com/openshift/windows-machine-config-operator/blob/4c09463d252e2bb20e133a41372be1d438267ba2/pkg/secrets/secrets.go#L54) by WMCO.

Reply:
That's correct. WMCO is agnostic to what GCP expects for it's meta-data key.

Reply:
I would say; Yes [100%](https://github.com/openshift/windows-machine-config-operator/blob/4c09463d252e2bb20e133a41372be1d438267ba2/pkg/secrets/secrets.go#L54) and defer to @openshift/openshift-team-windows-containers in case I'm missing something.

Reply:
> Are we providing any guarantee around the userData key being used in WMCO right now?

 If you're asking about a guarantee to users, we have no such guarantee. We maintain the userdata secret and do not allow users to change it. The user provides a private key in the form of another secret that they have control over, which is used when creating the userdata secret

Reply:
I want to know that if we are reading from the `userData` map key, that that will always work going forward and that it's not likely to be changed to something else by WMCO, I guess the key name forms part of the API contract at this point right?

Reply:
@JoelSpeed I don't see a need to ever change it, the key is what it is solely to interact with the Machine API. WMCO will not be changing the key name unless what is expected by MAPI is changed.

Reply:
+1

Reply:
i'll update this